File: DSRFQ.Web/Modules/ServerTypes/Costing/CostingPartCostingResultsColumns.ts
import { ColumnsBase, fieldsProxy } from '@serenity-is/corelib';
import { Column } from '@serenity-is/sleekgrid';
import { CostingPartCostingResultsRow } from './CostingPartCostingResultsRow';

export interface CostingPartCostingResultsColumns {
    Id: Column<CostingPartCostingResultsRow>;
    CostingPartId: Column<CostingPartCostingResultsRow>;
    CostingCategoryId: Column<CostingPartCostingResultsRow>;
    Name: Column<CostingPartCostingResultsRow>;
    Description: Column<CostingPartCostingResultsRow>;
    Quantity: Column<CostingPartCostingResultsRow>;
    DimensionUnitId: Column<CostingPartCostingResultsRow>;
    DimensionUnit: Column<CostingPartCostingResultsRow>;
    UnitPrice: Column<CostingPartCostingResultsRow>;
    CurrencyId: Column<CostingPartCostingResultsRow>;
    Total: Column<CostingPartCostingResultsRow>;
    IsTimeUnit: Column<CostingPartCostingResultsRow>;
    IsManual: Column<CostingPartCostingResultsRow>;
    InsertDate: Column<CostingPartCostingResultsRow>;
    InsertUserId: Column<CostingPartCostingResultsRow>;
    UpdateDate: Column<CostingPartCostingResultsRow>;
    UpdateUserId: Column<CostingPartCostingResultsRow>;
    DeleteDate: Column<CostingPartCostingResultsRow>;
    DeleteUserId: Column<CostingPartCostingResultsRow>;
    IsActive: Column<CostingPartCostingResultsRow>;
}

export class CostingPartCostingResultsColumns extends ColumnsBase<CostingPartCostingResultsRow> {
    static readonly columnsKey = 'Costing.CostingPartCostingResults';
    static readonly Fields = fieldsProxy<CostingPartCostingResultsColumns>();
}

File: DSRFQ.Web/Modules/Costing/CostingPartCostingResults/CostingPartCostingResultsColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace DSRFQ.Costing.Columns;

[ColumnsScript("Costing.CostingPartCostingResults")]
[BasedOnRow(typeof(CostingPartCostingResultsRow), CheckNames = true)]
public class CostingPartCostingResultsColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    public string CostingPartPartNumber { get; set; }
    public int CostingCategoryId { get; set; }
    [EditLink]
    public string Name { get; set; }
    public string Description { get; set; }
    public decimal Quantity { get; set; }
    public string DimensionUnitCode { get; set; }
    public string DimensionUnit { get; set; }
    public decimal UnitPrice { get; set; }
    public string CurrencyCode { get; set; }
    public decimal Total { get; set; }
    public bool IsTimeUnit { get; set; }
    public bool IsManual { get; set; }
    public DateTime InsertDate { get; set; }
    public string InsertBy { get; set; }
    public DateTime UpdateDate { get; set; }
    public string UpdateBy { get; set; }
    public DateTime DeleteDate { get; set; }
    public string DeleteBy { get; set; }
}